compiler/vm/vm_enums.nim

```diff
@@ -172,6 +172,39 @@ type
     avrNoLocation   ## Address points to valid VM memory. but not to the start
                     ## of a location
 
+  VmEventKind* = enum
+    # Here so the `ast_types.TAstDiag` can refer to them for VM Stace Traces
+    vmEvtOpcParseExpectedExpression
+    vmEvtUserError
+    vmEvtUnhandledException
+    vmEvtCannotCast
+    vmEvtCallingNonRoutine
+    vmEvtCannotModifyTypechecked
+    vmEvtNilAccess
+    vmEvtAccessOutOfBounds
+    vmEvtAccessTypeMismatch
+    vmEvtAccessNoLocation
+    vmEvtErrInternal
+    vmEvtIndexError
+    vmEvtOutOfRange
+    vmEvtOverOrUnderflow
+    vmEvtDivisionByConstZero
+    vmEvtArgNodeNotASymbol
+    vmEvtNodeNotASymbol
+    vmEvtNodeNotAProcSymbol
+    vmEvtIllegalConv
+    vmEvtMissingCacheKey
+    vmEvtCacheKeyAlreadyExists
+    vmEvtFieldNotFound
+    vmEvtNotAField
+    vmEvtFieldUnavailable
+    vmEvtCannotSetChild
+    vmEvtCannotAddChild
+    vmEvtCannotGetChild
+    vmEvtNoType
+    vmEvtTooManyIterations
```

Comment:
Had to move this here because of `traceReason` in the stacktrace, not exactly sure how to drop the tracereason business, just yet.

Reply:
As `traceReason` is never used (apart from initializing it), I think it's okay to just remove the field.

Reply:
Thanks, done.